### PR TITLE
ref(issues): make `ToggleButton` reusable across issue details

### PR DESCRIPTION
**Pull request by michellewzhang**
same behavior as before, just less copy paste now!

https://github.com/getsentry/sentry/assets/56095982/8b89ce38-cee6-4a17-909c-fefc981fa94a

### CODE REVIEW

File: static/app/views/issueDetails/toggleButton.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import styled from '@emotion/styled';
+
+import {Button} from 'sentry/components/button';
+import {t} from 'sentry/locale';
+
+interface IssueDetailsToggleButtonProps {
+  isShown: boolean;
+  setShownState: (boolean) => void;
```
Comment (scttcper) [inline static/app/views/issueDetails/toggleButton.tsx:8]: this is pretty nit picky - but props that take a function usually follow the `onClick` style. So maybe `onShownChange` or something
```suggestion
  onShownChange: (boolean) => void;
```

File: static/app/components/events/eventReplay/replayInlineOnboardingPanelBackend.tsx [hunk 2]
```diff
@@ -13,6 +13,7 @@ import {IconBroadcast} from 'sentry/icons/iconBroadcast';
 import {t, tct} from 'sentry/locale';
 import type {PlatformKey} from 'sentry/types';
 import {useReplayOnboardingSidebarPanel} from 'sentry/utils/replays/hooks/useReplayOnboarding';
+import ToggleButton from 'sentry/views/issueDetails/toggleButton';
```
Comment (vuluongj20) [inline static/app/components/events/eventReplay/replayInlineOnboardingPanelBackend.tsx:16]: Let's call this something like `SectionToggleButton` to be more specific about what it's for. "Toggle" to me only means "a switch between 2 states", and doesn't specifically connote "expand"/"collapse".

File: static/app/components/events/eventReplay/replayInlineOnboardingPanelBackend.tsx [hunk 3]
```diff
@@ -26,20 +27,16 @@ const OnboardingCTAButton = HookOrDefault({
 export default function ReplayInlineOnboardingPanelBackend({
   platform,
 }: OnboardingCTAProps) {
-  const [isHidden, setIsHidden] = useState(false);
+  const [isShown, setIsShown] = useState(true);
```
Comment (vuluongj20) [inline static/app/components/events/eventReplay/replayInlineOnboardingPanelBackend.tsx:30]: How about `expanded`/`collapsed`?

File: static/app/components/events/groupingInfo/index.tsx [hunk 4]
```diff
@@ -209,11 +209,7 @@ class GroupingInfo extends DeprecatedAsyncComponent<Props, State> {
       <EventDataSection
         type="grouping-info"
         title={t('Event Grouping Information')}
-        actions={
-          <ToggleButton onClick={this.toggle} priority="link">
-            {isOpen ? t('Hide Details') : t('Show Details')}
-          </ToggleButton>
-        }
+        actions={<ToggleButton isShown={isOpen} setShownState={this.toggle} />}
```
Comment (vuluongj20) [inline static/app/components/events/groupingInfo/index.tsx:212]: I'm assuming the `isOpen` state is stored across sessions?

Comment (michellewzhang) [reply static/app/components/events/groupingInfo/index.tsx:212]: that's what replay team is trying to push for (and what we're discussing at the meeting tomorrow). currently, the behavior isn't stored (and never has been)

File: static/app/components/events/eventReplay/replayInlineOnboardingPanelBackend.tsx (hunk 2 above)
Comment (michellewzhang) [reply static/app/components/events/eventReplay/replayInlineOnboardingPanelBackend.tsx:16]: ah that's my bad, i had the component named `IssueDetailsToggleButton` to be specific but forgot to import it as such - but can rename it to `SectionToggleButton`

File: static/app/views/issueDetails/sectionToggleButton.tsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+import {Button} from 'sentry/components/button';
+import {t} from 'sentry/locale';
+
+interface SectionToggleButtonProps {
+  isExpanded: boolean;
+  onExpandChange: (state: boolean) => void;
+}
+
+function SectionToggleButton({isExpanded, onExpandChange}: SectionToggleButtonProps) {
```
Comment (vuluongj20) [inline static/app/views/issueDetails/sectionToggleButton.tsx:11]: Let's also forward props to the button, so folks can add useful things like `aria-controls`.

**Review by vuluongj20 — APPROVED** [review]
Looks great! Thanks for doing this 🙏

**Pull request merged**